File: remove-linked-list-elements/remove-linked-list-elements.java
/**
 * Definition for singly-linked list.
 * public class ListNode {
 *     int val;
 *     ListNode next;
 *     ListNode() {}
 *     ListNode(int val) { this.val = val; }
 *     ListNode(int val, ListNode next) { this.val = val; this.next = next; }
 * }
 */
class Solution {
    public ListNode removeElements(ListNode head, int val) {
        ListNode temp = new ListNode(0, head);
        ListNode newList = temp;
        
        while(newList.next != null){
            if(newList.next.val == val){
                newList.next = newList.next.next;
            }
            else{
               newList = newList.next; 
            }
            
        }
        return temp.next;
    }
    void Print(ListNode head) {
    ListNode cur = head;
    while(cur!=null){
        System.out.println(cur.val);
        cur = cur.next;
    }
  
}
}

File: remove-linked-list-elements/remove-linked-list-elements.js
/**
 * Definition for singly-linked list.
 * function ListNode(val, next) {
 *     this.val = (val===undefined ? 0 : val)
 *     this.next = (next===undefined ? null : next)
 * }
 */
/**
 * @param {ListNode} head
 * @param {number} val
 * @return {ListNode}
 */
var removeElements = function(head, val) {
    let temp = new ListNode(0, head)
    let current = temp
    
    while(current.next != null){
        if(current.next.val == val){
            current.next = current.next.next
        }
        else {
            current = current.next
        }
    }
    return temp.next
};